tests/integration/workflows/java_gradle/test_java_gradle.py

```diff
@@ -36,7 +37,18 @@ class TestJavaGradle(TestCase):
 
     def setUp(self):
         self.artifacts_dir = tempfile.mkdtemp()
-        self.scratch_dir = tempfile.mkdtemp()
+
+        scratch_folder_override = None
+        if platform.system().lower() == "windows" and os.getenv("GITHUB_ACTIONS"):
+            # lucashuy: there is some really odd behaviour where
+            # gradle will refuse to work it is run within
+            # the default TEMP folder location in Github Actions
+            #
+            # use the runner's home directory as a workaround
+            scratch_folder_override = os.getenv("userprofile")
```

Comment:
Do we need to create another folder under `userprofile` so that test files won't clash with each other?

Reply:
Shouldn't need to, the `tempfile` library creates unique folder names each time that shouldn't clash with anything in the existing directory. We'll probably want to create a subfolder if this was a normal test, for a normal user just to keep things nice and tidy. But since this is the CI runner, I'm sure the runner won't mind